tests/networks.spec.ts

```diff
@@ -1,151 +1,299 @@
+        page,
+        "network.ovn.northbound_connection",
+      )) ?? "";
+    await updateSetting(
+      page,
+      "network.ovn.northbound_connection",
+      "text",
+      "foo",
+    );
+
+    await page.close();
+  });
+
+  test.afterAll(async ({ browser }) => {
+    const page = await browser.newPage();
+
+    await visitServerSettings(page);
+    await updateSetting(
+      page,
+      "network.ovn.northbound_connection",
+      "text",
+      initialNorthboundNetworkValue,
+    );
+
+    await page.close();
+  });
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto("/ui/");
+    await page.getByRole("link", { name: "Networks", exact: true }).click();
+    await page.getByRole("button", { name: "Create network" }).click();
+    await page.getByRole("heading", { name: "Create a network" }).click();
+    await page.getByLabel("Type").selectOption("OVN");
+    await page.getByLabel("Name").click();
+    await page.getByLabel("Name").fill(network);
+    await page.getByLabel("Uplink").selectOption({ index: 2 });
+  });
+
+  test("configure main OVN network settings", async ({ page }) => {
+    await activateAllTableOverrides(page);
+
+    await page
+      .getByPlaceholder("Enter description")
+      .fill(NETWORK_CONSTANTS.DESCRIPTION);
+
+    const radioButtons = await page.$$("input[type=radio]");
+    for (const button of radioButtons) {
+      await button.click({ force: true });
+    }
+
+    await page.getByLabel("IPv4 address").fill(NETWORK_CONSTANTS.IPV4_ADDRESS);
+    await page.getByLabel("IPv6 address").fill(NETWORK_CONSTANTS.IPV6_ADDRESS);
+
+    const options = ["false", "true"];
+    await checkAllOptions(page, "IPv4 NAT", options);
+    await checkAllOptions(page, "IPv6 NAT", options);
+  });
+
+  test("set OVN bridge MTU and hardware address", async ({ page }) => {
+    await visitNetworkConfiguration(page, "Bridge");
+
+    await setTextarea(page, "MTU", NETWORK_CONSTANTS.MTU);
+    await setTextarea(
+      page,
+      "Hardware address",
+      NETWORK_CONSTANTS.HARDWARE_ADDRESS,
+    );
+  });
+
+  test("configure OVN network DNS settings", async ({ page }) => {
+    await visitNetworkConfiguration(page, "DNS");
+
+    await setTextarea(page, "DNS domain", NETWORK_CONSTANTS.DNS_DOMAIN);
+    await setTextarea(page, "DNS search", NETWORK_CONSTANTS.DNS_SEARCH);
+  });
+
+  test("set OVN network IPv4 DHCP and L3 options", async ({ page }) => {
+    await visitNetworkConfiguration(page, "IPv4");
+
+    const options = ["false", "true"];
+    await checkAllOptions(page, "IPv4 DHCP", options);
+    await checkAllOptions(page, "IPv4 L3 only", options);
+  });
+
+  test("configure OVN network IPv6 DHCP and L3 settings", async ({ page }) => {
+    await visitNetworkConfiguration(page, "IPv6");
+
+    const options = ["false", "true"];
+    for (const option of options) {
+      await page
+        .getByRole("combobox", { name: "IPv6 DHCP", exact: true })
+        .selectOption(option);
+    }
```

Comment:
"IPv6 DHCP" also matches "IPv6 DHCP stateful". I haven't enforced "exact: true" at the function "checkAllOptions" since most helper functions don't have it.  That's the reason I made the exception and used a for loop.

Reply:
Right, makes sense. Thanks for clarifying.